utils/node_local_test.py

```diff
@@ -2386,77 +2383,159 @@ def test_cont_clone(self):
+
+    def __init__(self):
+        self._stdout = sys.stdout
+        self._outputs = {}
+        sys.stdout = self
+
+    def write(self, value):
+        """Print to stdout.  If this is the main thread then print it, else save it"""
+
+        thread = threading.current_thread()
+        if not thread.daemon:
+            self._stdout.write(value)
+            return
+        thread_id = thread.ident
+        try:
+            self._outputs[thread_id] += value
+        except KeyError:
+            self._outputs[thread_id] = value
+
+    def sprint(self, value):
+        """Really print something to stdout"""
+        self._stdout.write(value + '\n')
+
+    def get_thread_output(self):
+        """Return the stdout by the calling thread, and reset for next time"""
+        thread_id = threading.get_ident()
+        data = self._outputs[thread_id]
+        del self._outputs[thread_id]
+        return data
+
+    def flush(self):
+        """Flush"""
+        self._stdout.flush()
+
+    def __del__(self):
+        sys.stdout = self._stdout
+
 def run_posix_tests(server, conf, test=None):
     """Run one or all posix tests
 
     Create a new container per test, to ensure that every test is
     isolated from others.
     """
 
-    def _run_test():
-        pt.call_index = 0
+    def _run_test(ptl=None, function=None, test_cb=None):
+        ptl.call_index = 0
         while True:
-            pt.needs_more = False
-            pt.test_name = fn
+            ptl.needs_more = False
+            ptl.test_name = function
             start = time.time()
-            print('Calling {}'.format(fn))
+            out_wrapper.sprint('Calling {}'.format(function))
+            print('Calling {}'.format(function))
+
+            # Do this with valgrind disabled as this code is run often and valgrind has a big
+            # performance impact.  There are other tests that run with valgrind enabled so this
+            # should not reduce coverage.
             try:
-                # Do this with valgrind disabled as this code is run often and valgrind has a big
-                # performance impact.  There are other tests that run with valgrind enabled so this
-                # should not reduce coverage.
-                pt.container = create_cont(conf,
-                                           pool.id(),
-                                           ctype="POSIX",
-                                           valgrind=False,
-                                           log_check=False,
-                                           label=fn)
-                pt.container_label = fn
-                rc = obj()
-                destroy_container(conf,
-                                  pool.id(),
-                                  pt.container_label,
-                                  valgrind=False,
-                                  log_check=False)
-                pt.container = None
+                ptl.container = create_cont(conf,
```

Comment:
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'log_parse' in function call